### ISSUE TITLE
vectormap::find() is easy to use incorrectly

### ISSUE THREAD

**Issue by jwpeterson** [libMesh/libmesh#1281]
In the course of working on #1278, I wrote some code like:
```
if (global_index_map.find(neighbor->id()) == global_index_map.end())
  continue;
```
which I *thought* would skip ids not found in the `vectormap` called `global_index_map`.  Instead, it turns out that `vectormap::find()` simply returns the result of a `std::lower_bound` call on a vector, i.e. an iterator to the first element in the range which is *not less than* the searched-for value -- completely different behavior than `map::find()`.

I'd like to fix this so that `vectormap::find()` works similarly to `map::find()` instead.

**Comment by jwpeterson** [comment]
Aaand I added this stupid code in bdbc50675dfd10bcf4f44554009aef18781f1a0a.  Super.

**Comment by roystgnr** [comment]
Ha!  It wasn't my fault?!

**Comment by pbauman** [comment]
The existing interface is not entirely crazy, depending on what `vectormap` is supposed to do (I haven't looked): http://www.cplusplus.com/reference/map/multimap/find/

**Comment by jwpeterson** [comment]
We sort/uniquify the vector, so vectormap is supposed to behave more like std::map.

**Comment by jwpeterson** [comment]
Fixed by #1282.
